### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Nordexz**
[DEMO LINK](https://nordexz.github.io/react_list-of-goods-hooks/)

### CODE REVIEW

**Review by arkushenko-anna — APPROVED** [review]
GJ!

File: src/App.tsx [hunk 1]
```diff
@@ -15,48 +16,123 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
```
Comment (sergii-nosachenko) [inline src/App.tsx:22]: Add values to enum

File: src/App.tsx [hunk 2]
```diff
@@ -15,48 +16,123 @@ export const goodsFromServer = [
...
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
```
Comment (sergii-nosachenko) [inline src/App.tsx:33]: Add return type

File: src/App.tsx [hunk 3]
```diff
@@ -15,48 +16,123 @@ export const goodsFromServer = [
...
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((g1, g2) => {
+    switch (sortType) {
+      case SortType.NONE:
+        return 0;
```
Comment (sergii-nosachenko) [inline src/App.tsx:39]: Combine it with default case

File: src/App.tsx [hunk 4]
```diff
@@ -15,48 +16,123 @@ export const goodsFromServer = [
...
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReversed] = useState(false);
```
Comment (sergii-nosachenko) [inline src/App.tsx:57]:
```suggestion
  const [isReversed, setIsReversed] = useState(false);
```

File: src/App.tsx [hunk 5]
```diff
@@ -15,48 +16,123 @@ export const goodsFromServer = [
...
+  const sortByType = (sortingType: SortType) => {
+    setSortType(sortingType);
+  };
+
+  const reset = () => {
+    setReversed(false);
+    setSortType(SortType.NONE);
+  };
+
+  const rstBttn = sortType !== SortType.NONE || isReversed;
```
Comment (sergii-nosachenko) [inline src/App.tsx:73]: What is `rstBttn`? Name it as a normal boolean value.

File: src/App.tsx [hunk 6]
```diff
@@ -15,48 +16,123 @@ export const goodsFromServer = [
...
+  const orderingGoods = getReorderedGoods(
```
Comment (sergii-nosachenko) [inline src/App.tsx:74]: This constant is not a process, use a proper name

File: src/App.tsx [hunk 7]
```diff
@@ -15,48 +16,123 @@ export const goodsFromServer = [
...
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {orderingGoods
+          .map(
+            word => <li data-cy="Good" key={word}>{word}</li>,
+          )}
       </ul>
```
Comment (sergii-nosachenko) [inline src/App.tsx:136]: Create a separate component for the list

File: src/App.tsx [hunk 8]
```diff
@@ -15,48 +16,123 @@ export const goodsFromServer = [
...
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {orderingGoods
+          .map(
+            word => <li data-cy="Good" key={word}>{word}</li>,
```
Comment (sergii-nosachenko) [inline src/App.tsx:134]: Split into rows

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 9]
```diff
@@ -15,49 +17,118 @@ export const goodsFromServer = [
...
+enum SortType {
+  NONE = 'NONE',
+  ALPHABET = 'ALPHABET',
+  LENGTH = 'LENGTH',
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
```
Comment (Andr1yK) [inline src/App.tsx:29]:
```typescript
interface Person {
  name: string;
  age: number;
}

const person: Person = { name: 'Alice', age: 30 };
```

In TypeScript, the `interface` keyword is used to define object types. While the `type` keyword can also be used to define object types, `interface` has some advantages over `type` in certain cases. Here are some rules to consider when choosing between `interface` and `type` for typing objects in TypeScript:

1. **Prefer `interface` for object types:** The `interface` keyword is the preferred way to define object types in TypeScript. This is because interfaces can be extended, while types cannot. Interfaces can also be used to define callable objects, which is not possible with types.

2. **Use `type` for union types:** The `type` keyword is useful for defining union types, where a variable can have multiple possible types. This can be useful in cases where you want to define a variable that can either be a string or a number, for example.

3. **Use `interface` for extending:** If you need to extend an object type, you should use an interface. Interfaces can be extended with the `extends` keyword, which allows you to add or modify properties of an existing interface.

4. **Be consistent:** It's important to be consistent in your codebase when choosing between `interface` and `type` for object types. If you're working on a project with existing code, follow the established conventions. If you're starting a new project, choose one and stick with it.

In summary, while both `interface` and `type` can be used for typing objects in TypeScript, `interface` is the preferred way to define object types, especially when you need to extend an existing interface. The `type` keyword is useful for defining union types, but should be used sparingly for object types.

File: src/App.tsx [hunk 10]
```diff
@@ -15,49 +17,118 @@ export const goodsFromServer = [
...
+  const sortByType = (sortingType: SortType) => {
+    setSortType(sortingType);
+  };
+
+  const reset = () => {
+    setIsReversed(false);
+    setSortType(SortType.NONE);
+  };
+
+  const isResetButton = sortType !== SortType.NONE || isReversed;
```
Comment (Andr1yK) [inline src/App.tsx:73]:
```suggestion
  const isResetButtonVisible = sortType !== SortType.NONE || isReversed;
```

File: src/App.tsx (hunk 6 above)
Comment (Andr1yK) [reply src/App.tsx:74]: let's fix

**Review by Andr1yK — APPROVED** [review]
